lib/crypto_env_var.rb

```diff
@@ -4,7 +4,25 @@
 
 
 module CryptoEnvVar
+  CRYPTO_ENV_VAR  = "CRYPTO_ENV"
+  DECRYPT_KEY_VAR = "CRYPTO_ENV_DECRYPT_KEY"
+  CRYPTO_ENV      = lambda { ENV.fetch(CRYPTO_ENV_VAR) }
+  DECRYPT_KEY     = lambda { ENV.fetch(DECRYPT_KEY_VAR) }
+
+
   class << self
+    def bootstrap!(read_from: CRYPTO_ENV, decrypt_with: DECRYPT_KEY, override_env: true)
```

Comment:
Maybe allow passing `env` as an optional argument? This would make testing easier. The function signature could then be:

```ruby
def bootstrap!(read_from: CRYPTO_ENV, decrypt_with: DECRYPT_KEY, env: ENV, override_env: true)
```

Reply:
I thought about that, and it would indeed help with the `ENV[key] = value` bit in the loop on line 21.

It wouldn't be so straightforward for the `CRYPTO_ENV` and `DECRYPT_KEY` callables, though. I mean, I could pass the `env` to the callables when invoking them in `read_value`, but:
* Some callable strategies might not need to access the `ENV` at all, so it would be odd to pass it.
* If the `ENV` or another object is always passed as an argument, then this means that all user-provided callables need to accept it. That's ok for Procs, as arguments are checked with block semantics, but custom objects and lambdas will check the number of arguments and must explicitly declare it. It doesn't seem very elegant.

I don't have anything against dependency injection, but in this case it seems to only allow for a 50%-solution, so I preferred to not do it all. Does it make sense?